pkg/compactor/retention/expiration.go

```diff
@@ -132,6 +133,43 @@ func NewTenantsRetention(l Limits) *TenantsRetention {
 	}
 }
 
+func (tr *TenantsRetention) PolicyFor(userID string, lbs labels.Labels) string {
+	policyStreamMapping := tr.limits.PoliciesStreamMapping(userID)
+	var (
+		matchedRule       *validation.PriorityStream
+		matchedPolicyName string
+		found             bool
+	)
+Outer:
+	for policyName, policyStreams := range policyStreamMapping {
+		for _, policyStream := range policyStreams {
+			for _, m := range policyStream.Matchers {
+				if !m.Matches(lbs.Get(m.Name)) {
+					continue Outer
+				}
+			}
+			// the rule is matched.
+			if found {
+				// if the current matched rule has a higher priority we keep it.
+				if matchedRule.Priority > policyStream.Priority {
+					continue
+				}
+				// if priority is equal we keep the lowest retention.
```

Comment:
Isn't this condition the same as the previous?

Reply:
same: I changed the whole code so this isn't a thing anymore